core/src/main/java/com/adobe/aem/guides/wknd/core/services/PageSearchServiceImpl.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Created on 09-08-2023 21:34 by jjedlins
+ *
+ * Copyright (c) 2001-2023 Unity S.A.
+ * ul. Strzegomska 2-4, 53-611 Wrocław, Poland
+ * Wszelkie prawa zastrzeżone
+ *
+ * Niniejsze oprogramowanie jest własnością Unity S.A.
+ * Wykorzystanie niniejszego oprogramowania jest możliwe tylko na podstawie
+ * i w zgodzie z warunkami umowy licencyjnej zawartej z Unity S.A.
+ */
+
+package com.adobe.aem.guides.wknd.core.services;
+
+import java.awt.*;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+import java.util.Spliterator;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import javax.jcr.Node;
+import javax.jcr.RepositoryException;
+import javax.jcr.query.Query;
+
+import org.apache.commons.lang3.builder.ToStringBuilder;
+import org.apache.jackrabbit.webdav.property.ResourceType;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.metatype.annotations.Designate;
+
+import com.day.cq.wcm.api.Page;
+
+
+@Component(service = PageSearchService.class, immediate = true)
+@Designate(ocd = PageSearchConfig.class)
+public class PageSearchServiceImpl implements PageSearchService {
+
+
+	private String rootPath;
+
+	@Activate
+	public void activate(PageSearchConfig pageSearchConfig) {
+		this.rootPath = pageSearchConfig.getRootPath();
+	}
+
+	@Override
+	public String getRootPath() {
+
+		return rootPath;
+	}
+
+	@Override
+	public List<PageSearchResultData> executeQuery(String rootPath, String parameterFromUrl, int maximumNumberOfResults, ResourceResolver resourceResolver) {
+
+		String query = "SELECT page.* FROM [cq:Page] AS page "
```

Comment:
I tested this query locally and it looks overcomplicated and also doesnt look like it's working properly.
Try using relative property path to simplify it.
Something like:
"SELECT * FROM [cq:Page] WHERE ISDESCENDANTNODE('%s') AND [jcr:content/jcr:title] LIKE '%%s%'"
there is also SQL2 LOWER directive which could solve potential case sensitivity issues

Reply:
Yea, I kinda do not understand this SQL2 and all the documentation is shitty. I was just thinking how to do something from SQL in SQL2. That's why I did this join. And now I remember that I wanted to ask something - how to limit number of results directly in SQL2? In SQL I would just put `LIMIT 2;` at the end of SQL :D